pkg/core/resources/apis/mesh/traffic_log_validator.go

```diff
@@ -0,0 +1,42 @@
+package mesh
+
+import (
+	"fmt"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/pkg/core/validators"
+)
+
+func (d *TrafficLogResource) Validate() error {
+	var err validators.ValidationError
+	err.Add(d.validateSources())
+	err.Add(d.validateDestinations())
+	return err.OrNil()
+}
+
+func (d *TrafficLogResource) validateSources() validators.ValidationError {
+	return ValidateSelectors(validators.RootedAt("sources"), d.Spec.Sources, ValidateSelectorOpts{})
+}
+
+func (d *TrafficLogResource) validateDestinations() (err validators.ValidationError) {
+	return ValidateSelectors(validators.RootedAt("destinations"), d.Spec.Destinations, ValidateSelectorOpts{
```

Comment:
Seems to be identical to `TrafficRoute`.

Let's refactor out into a reusable func.